front/src/applications/stdcm/views/StdcmRequestModal.tsx

```diff
@@ -48,71 +54,77 @@ export default function StdcmRequestModal(props) {
   // https://developer.mozilla.org/en-US/docs/Web/API/AbortController
   const controller = new AbortController();
 
-  const timetableId = osrdconf.timetableID;
+  const timetableId = osrdconf.timetableID?.toString();
 
   // Returns a promise that will be a fetch or an axios (through react-query)
   const stdcmRequest = async () => {
-    const params = formatStdcmConf(dispatch, setFailure, t, osrdconf);
-    return postStdcm(params).unwrap();
+    const params = formatStdcmConf(dispatch, setFailure, t, osrdconf as OsrdStdcmConfState);
+    return postStdcm(params as PostStdcmApiArg).unwrap();
   };
 
   useEffect(() => {
     if (currentStdcmRequestStatus === STDCM_REQUEST_STATUS.pending && timetableId) {
       stdcmRequest()
         .then((result) => {
           setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.success);
-          dispatch(updateItinerary(result.path));
-
-          const fakedNewTrain = {
-            ...cloneDeep(result.simulation),
-            id: 1500,
-            isStdcm: true,
-          };
-
-          fakedNewTrain.base.stops = fakedNewTrain.base.head_positions[0].map(
-            (headPosition, index) => ({
-              id: fakedNewTrain.id,
-              name: `stop ${index}`,
-              time: headPosition.time,
-              position: headPosition.position,
-              duration: 0,
+          if (('path' && 'simulation') in result) {
```

Comment:
this does not work,
'path' && 'simulation' evaluates as 'simulation'
`('path' && 'simulation') in { 'simulation': {}  }` will be true
`A && B` returns B if A is truthy, otherwise it returns A

Reply:
need to talk about it

Reply:
done